### PR TITLE
New Package: gavl-1.4.0

### PR DESCRIPTION
**Pull request by logenkain**
I'm still not happy with the only_for_arches, but the config script can't be cross compiled.  If anyone has ideas, let me know.

### CODE REVIEW

**Comment by pullmoll** [comment]
I'd suggest `nocross=yes` and if there are issues with musl libc, fix them by adding appropriate patches.

**Comment by logenkain** [comment]
On Tue, Sep 29, 2015 at 11:21:00PM -0700, Jürgen Buchmüller wrote:
> I'd suggest `nocross=yes` and if there are issues with musl libc, fix them by adding appropriate patches.
> 

I suppose I should learn how to do that.

**Comment by logenkain** [comment]
Actually, if someone wants to point me in a direction to learn this... I don't even know how to test musl without using travis.

**Comment by xtraeme** [comment]
https://github.com/voidlinux/void-packages#building-packages-natively-for-the-musl-c-library

Start from this part:

```
Wait until all packages are built and when ready, prepare a new masterdir with the musl packages:
```

File: srcpkgs/gavl/template [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+distfiles="${SOURCEFORGE_SITE}/gmerlin/${pkgname}/${version}/${pkgname}-${version}.tar.gz"
+checksum=51aaac41391a915bd9bad07710957424b046410a276e7deaff24a870929d33ce
+
+post_install(){
+	ln -sf libgavl.so.${version} ${DESTDIR}/usr/lib/libgavl.so.1
+}
+
+
+gavl-devel_package() {
+	depends="gavl>=${version}_${revision} $hostmakedepends"
```
Comment (xtraeme) [inline srcpkgs/gavl/template]: depending on `hostmakedepends` is wrong, please remove this.

**Comment by logenkain** [comment]
On Wed, Sep 30, 2015 at 07:40:57AM -0700, Juan RP wrote:
> Start from this part:
> 
> ```
> Wait until all packages are built and when ready, prepare a new masterdir with the musl packages:
> ```

Alright, thanks, I'll try to look into that while I have time.  I
suppose I should reread the guides, it's probably been about a year
since I've read them last and things have changed.

**Comment by logenkain** [comment]
Well, I get this "benchmark.c:1850:3: error: unknown type name 'cpu_set_t'"

I really don't know what to do about that.  Tried searching around and found other people having issues with that and musl, but, that's it.  I'm not sure what to do with it.

**Comment by pullmoll** [comment]
#### For a hint here's what I do in such a case

##### Does the function or type exist in musl libc?
```
$ grep -r cpu_set_t masterdir-musl/usr/include
masterdir-musl/usr/include/pthread.h:struct cpu_set_t;
masterdir-musl/usr/include/pthread.h:int pthread_getaffinity_np(pthread_t, size_t, struct cpu_set_t *);
masterdir-musl/usr/include/pthread.h:int pthread_setaffinity_np(pthread_t, size_t, const struct cpu_set_t *);
masterdir-musl/usr/include/sched.h:typedef struct cpu_set_t { unsigned long __bits[128/sizeof(long)]; } cpu_set_t;

```
It does, so the file which does not build seems to lack the required include.

##### Create a patch
```
$ cp masterdir-musl/builddir/gavl-1.4.0/src/benchmark.c .
$ vi benchmark.c
[ add the #include <pthread.h> somewhere ]
$ mkdir -p srcpkgs/gavl-1.4.0/patches
$ diff -ru masterdir-musl/builddir/gavl-1.4.0/src/benchmark.c benchmark.c > srcpkgs/gavl-1.4.0/patches/musl-pthread.patch
$ vi srcpkgs/gavl-1.4.0/patches/musl-pthread.patch
[ make the patch usable with the default -p0, i.e. first level is a subdirectory of masterdir-musl/builddir/gavl-1.4.0 ]
```

##### Build package again
```
$ ./xbps-src -m masterdir-musl clean
$ ./xbps-src -m masterdir-musl pkg gavl-1.40
```

Repeat until the build succeeds.

**Comment by logenkain** [comment]
On Thu, Oct 01, 2015 at 01:50:48AM -0700, Jürgen Buchmüller wrote:
> #### For a hint here's what I do in such a case
> 
> ##### Does the function or type exist in musl libc?
> ```
> $ grep -r cpu_set_t masterdir-musl/usr/include
> masterdir-musl/usr/include/pthread.h:struct cpu_set_t;
> masterdir-musl/usr/include/pthread.h:int pthread_getaffinity_np(pthread_t, size_t, struct cpu_set_t *);
> masterdir-musl/usr/include/pthread.h:int pthread_setaffinity_np(pthread_t, size_t, const struct cpu_set_t *);
> masterdir-musl/usr/include/sched.h:typedef struct cpu_set_t { unsigned long __bits[128/sizeof(long)]; } cpu_set_t;
> 
> ```
> It does, so the file which does not build seems to lack the required include.
> 
> ##### Create a patch
> ```
> $ cp masterdir-musl/builddir/gavl-1.4.0/src/benchmark.c .
> $ vi benchmark.c
> [ add the #include <pthread.h> somewhere ]
> $ mkdir -p srcpkgs/gavl-1.4.0/patches
> $ diff -ru masterdir-musl/builddir/gavl-1.4.0/src/benchmark.c benchmark.c > srcpkgs/gavl-1.4.0/patches/musl-pthread.patch
> $ vi srcpkgs/gavl-1.4.0/patches/musl-pthread.patch
> [ make the patch usable with the default -p0, i.e. first level is a subdirectory of masterdir-musl/builddir/gavl-1.4.0 ]
> ```
> 
> ##### Build package again
> ```
> $ ./xbps-src -m masterdir-musl clean
> $ ./xbps-src -m masterdir-musl pkg gavl-1.40
> ```
> 
> Repeat until the build succeeds.
> 

Thanks a lot, hopefully I'll have time to work on it tonight.  I might
have to make a copy of that and post it on the forums (or perhaps the
wiki?) Always good to see what people do when they run into snags.

**Comment by pullmoll** [comment]
This example was very specific and probably isn't much of a help for the general case.
What it needs to fix musl libc issues is a little background knowledge of what a libc does and where the differences between GNU libc and musl libc are.

This information can most of the times be found in other templates and associated patches for common pitfalls.

Generally speaking fixing musl libc issues is probably nothing for the absolute beginner. You can always leave musl errors in the travis-ci logs and ask other developers to take a look. Or it can be merged and musl libc issues can be taken care of afterwards.

**Comment by logenkain** [comment]
Yea I've been messing around and haven't figured it out yet.  So just put in no cross and leave it at that then?

**Comment by pullmoll** [comment]
I would approve that. You need to rebase anyway, because meanwhile `common/shlibs` has changed.
I can take a look at the package and try to make it work with musl

**Pull request merged**